feat: add flux dapp integration

gguuttss opened the pull request:
## Description
This PR adds Flux Protocol dApp integration to the library, providing functionality to query CDPs and Reservoir positions.

### Features
- **CDP Queries**: Get detailed information at specified state version about user CDPs including:
  - Collateral amounts
  - Debt amounts
  - Interest rates
  - Position status
  - Privileged borrower status

- **Reservoir Queries**: Get users' reservoir (aka stability pool) positions at specified state version including:
  - Pool token balances
  - Collateral and FUSD values of those pool tokens
  - Total pool tokens in circulation
  - Value of 1 pool token

### Technical Details
- Uses `BigNumber` for precise decimal handling
- Efficient API call batching as efficient as possible (I think ha)
- Provides type-safe schemas and constants
- Follows the library's established patterns for dapp integrations as best I could

### Testing
- Includes test coverage that demonstrate proper usage of the new functionality

### Files Added
- `getFluxCdps.ts` - CDP query implementation
- `getFluxReservoir.ts` - Reservoir query implementation
- `schemas.ts` - Type definitions and schemas
- `constants.ts` - Protocol constants and addresses
- `index.ts` - Public exports
- Test files for both main features

## Future improvements
- Abstract away pool token logic (get total pool token supply, get value of 1 pool token, etc.), seems like this would be useful for many other dApps as well.

gguuttss committed:
feat: add flux dapp integration

Add Flux Protocol dapp integration to effects with:
- CDP queries (get accounts' debts at certain state version and more)
- Reservoir position queries (get accounts' reservoir deposits, and reservoir info)
- Number handling with BigNumber
- Batching of API calls

xstelea commented on packages/effects/src/dapps/flux/constants.ts:
```diff
@@ -0,0 +1,30 @@
+  collaterals: {
+    xrd: {
+      collateralAddress:
+        "resource_rdx1tknxxxxxxxxxradxrdxxxxxxxxx009923554798xxxxxxxxxradxrd",
```
use `import { Assets } from "../../assets/constants";` to reference XRD

xstelea commented on packages/effects/src/dapps/flux/constants.ts:
```diff
@@ -0,0 +1,30 @@
+    },
+    lsulp: {
+      collateralAddress:
+        "resource_rdx1thksg5ng70g9mmy9ne7wz0sc7auzrrwy7fmgcxzel2gvp8pj0xxfmf",
```
use CaviarNineConstants to reference LSULP

xstelea commented on packages/effects/src/dapps/flux/getFluxCdps.ts:
```diff
@@ -0,0 +1,282 @@
+          keyValueStoreDataService({
+            key_value_store_address: FluxConstants.xrdKvsAddress,
+            keys: xrdKeys.items.map((k: any) => ({
+              key_json: k.key.programmatic_json,
```
you don't have to use any here

xstelea commented on packages/effects/src/dapps/flux/getFluxCdps.ts:
```diff
@@ -0,0 +1,282 @@
+          keyValueStoreDataService({
+            key_value_store_address: FluxConstants.lsulpKvsAddress,
+            keys: lsulpKeys.items.map((k: any) => ({
+              key_json: k.key.programmatic_json,
```
remove any

xstelea commented on packages/effects/src/dapps/flux/getFluxCdps.ts:
```diff
@@ -0,0 +1,282 @@
+              : lsulpInterestData.entries;
+
+          const interestInfo = interestData.find((entry: any) => {
+            const nodeData = entry.value.programmatic_json;
+            return (
+              nodeData &&
+              nodeData.kind === "Tuple" &&
+              new BigNumber(nodeData.fields[0].value).isEqualTo(interestRate)
+            );
+          });
```
something like this to avoid `any`
```
const interestInfo = interestData.find((entry) => {
const nodeData = entry.value.programmatic_json;
if (nodeData.kind === "Tuple") {
    const tupleFields = nodeData.fields;
    if (tupleFields[0] && tupleFields[0].kind === "Decimal") {
        const interestRateValue = new BigNumber(tupleFields[0].value);
        return interestRateValue.isEqualTo(interestRate);
    }
}
});
```

xstelea reviewed (approved):
LGTM. added some comments to remove any types and use already defined constants. feel free to update if you want

gguuttss committed:
chore: applied suggestions from PR review

- removed unnecessary any types
- used already existing LSULP and XRD resource address constants

gguuttss commented:
Thanks for reviewing. Good suggestions, applied them.

gguuttss commented:
> LGTM. added some comments to remove any types and use already defined constants. feel free to update if you want

Thanks for reviewing. Good suggestions, applied them.
